### PR TITLE
added editorconfig

### PR DESCRIPTION
**Pull request by caarlos0**
Added `.editorconfig` file, so it's easier to avoid mistakes (like spaces vs tabs, etc).

### CODE REVIEW

**Comment by vdemeester** [comment]
Why not but pretty much all Go editors are "wired" to gofmt (on save) and/or should be (otherwise the build fail here anyway), so it's more a nice-to-have than a requirement.

/cc @thaJeztah @dnephin @justincormack

**Comment by caarlos0** [comment]
@vdemeester yeap, but there are also shell scripts and other kinds of files... I thought it would be useful to have them all under the same kind of fomating...

**Comment by dnephin** [comment]
IMO what is more valuable is a tool to enforce style as a git hook. I use http://pre-commit.com/ with https://gist.github.com/dnephin/c7776ecb5cdcf02eb1a6 to ensure that formatting is correct.

If we're going to add this, it's probably more appropriate under `contrib/` somewhere, instead of the root of the repo. We can add `.editorconfig` to `.gitignore` so that developers can customize their copy as they want.

**Comment by thaJeztah** [comment]
I'm wondering if this falls in the same category as https://github.com/docker/docker/blob/389d5cde7b5474cd33d251aadc96be1f0bdbc711/.gitignore#L2-L3

**Comment by dnephin** [comment]
True, we don't need to add it to `.gitignore`.

**Comment by tianon** [comment]
Yeah, I agree -- this is useful information, and it's nice to see folks trying to standardize the way it's communicated, but I don't think the standard is firm enough yet that committing this at the top-level is warranted (perhaps more prudent once/if more editors start supporting this file out-of-the-box or if it becomes more common).  `contrib/` seems like a great place for it (`contrib/editorconfig`, for example, so that it's not hiding and is easy to find and symlink to `.editorconfig` if folks want to use it via something simple like `ln -s contrib/editorconfig .editorconfig`).

**Comment by caarlos0** [comment]
@tianon @thaJeztah @justincormack @vdemeester  Done, what do you guys think?

File: .gitignore [hunk 1]
```diff
@@ -29,3 +29,4 @@ man/man1
 man/man5
 man/man8
 vendor/pkg/
+.editorconfig
```
Comment (thaJeztah) [inline .gitignore]: Can you move this up, to keep the list in alphabetical order?

**Review by thaJeztah — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (caarlos0) [inline .gitignore]: done!

File: contrib/editorconfig [hunk 2]
```diff
@@ -0,0 +1,9 @@
+root = true
+
+[*]
```
Comment (thaJeztah) [inline contrib/editorconfig]: Actually, just noticing this; should this be limited to `.go` files? We use spaces for Markdown (`.md`)

Comment (caarlos0) [inline contrib/editorconfig]: @thaJeztah Just added the missing `[*.md]` section.

Comment (vdemeester) [inline contrib/editorconfig]: We do use tab for shell script though :angel:, I think it's good as is. `*.md` is handled down :stuck_out_tongue_closed_eyes:

**Review by vdemeester — APPROVED** [review]
LGTM 🐸

**Comment by tianon** [comment]
LGTM :+1:

**Review by thaJeztah — APPROVED** [review]
LGTM, thanks!

**Comment by dnephin** [comment]
I just learned something that changed my opinion on this. github actually supports `.editorconfig`! So if we set 

```
[*]
indent_size = 4
```
Viewing code and diffs will show 4 space tabs instead of 8 space tabs. 

I would be in favor of moving this out of contrib and back into the root of the repo so that we can have nicer looking code and diffs on github.

**Comment by thaJeztah** [comment]
@dnephin that's nice! Is it only indent size that's taken into account, no other "side effects"?

**Comment by dnephin** [comment]
I'm not sure, there doesn't seem to be much information out there about it. http://stackoverflow.com/a/33899831/444646 suggests that they never officially announced the support.

I suspect it does support any options that make sense from the UI. Maybe they apply to the in-browser editor as well.

**Comment by tianon** [comment]
Honestly, I kinda like that GitHub shows tabs at 8 spaces because it makes it much more obvious when someone screws it up in a PR because the 4-real-space-indents don't line up properly. :innocent:

**Comment by thaJeztah** [comment]
Oh, you've got a point, it has helped me as well (if only GitHub had a `show whitespace` option)

**Comment by dnephin** [comment]
Shouldn't we have automated style checks for that instead of relying on a human to notice it?

http://pre-commit.com/ with https://gist.github.com/dnephin/c7776ecb5cdcf02eb1a6 is a great option.

**Comment by thaJeztah** [comment]
That would be good, however, I can imagine there's combinations that are harder to validate strictly

**Pull request merged**